diff -uNr initscripts-9.03.17__orig/rc.d/rc.sysinit initscripts-9.03.17/rc.d/rc.sysinit
--- initscripts-9.03.17__orig/rc.d/rc.sysinit	2010-09-01 14:34:41.000000000 +0000
+++ initscripts-9.03.17/rc.d/rc.sysinit	2011-06-07 23:23:46.000000000 +0000
@@ -100,6 +100,12 @@
  [ "$BOOTUP" = "color" ] && echo -en "\\033[0;39m"
  PRODUCT=$(sed "s/Fedora \(.*\) \?release.*/\1/" /etc/system-release)
  echo " $PRODUCT"
+elif [[ "$system_release" =~ "CentOS" ]]; then
+ [ "$BOOTUP" = "color" ] && echo -en "\\033[0;36m"
+ echo -en "CentOS"
+ [ "$BOOTUP" = "color" ] && echo -en "\\033[0;39m"
+ PRODUCT=$(sed "s/CentOS \(.*\) \?release.*/\1/" /etc/system-release)
+ echo " $PRODUCT"
 else
  PRODUCT=$(sed "s/ release.*//g" /etc/system-release)
  echo "$PRODUCT"
