diff -up initscripts-9.49.24/rwtab.peerdns initscripts-9.49.24/rwtab
--- initscripts-9.49.24/rwtab.peerdns	2015-01-15 01:57:03.000000000 -0700
+++ initscripts-9.49.24/rwtab	2015-04-03 09:55:58.310658951 -0600
@@ -25,6 +25,7 @@ empty	/var/tmp
 files	/etc/adjtime
 files	/etc/ntp.conf
 files	/etc/resolv.conf
+files	/etc/resolv-peerdns.conf
 files	/etc/lvm/cache
 files	/etc/lvm/archive
 files	/etc/lvm/backup
diff -up initscripts-9.49.24/sysconfig/network-scripts/ifup-post.peerdns initscripts-9.49.24/sysconfig/network-scripts/ifup-post
--- initscripts-9.49.24/sysconfig/network-scripts/ifup-post.peerdns	2015-01-15 01:57:03.000000000 -0700
+++ initscripts-9.49.24/sysconfig/network-scripts/ifup-post	2015-04-03 09:55:58.313658959 -0600
@@ -26,12 +26,12 @@ fi
 if [ "$PEERDNS" != "no" ] ||[ -n "$RESOLV_MODS" -a "$RESOLV_MODS" != "no" ]; then
   [ -n "$MS_DNS1" ] && DNS1=$MS_DNS1
   [ -n "$MS_DNS2" ] && DNS2=$MS_DNS2
-  if [ -n "$DNS1" ] && ! grep -q "^nameserver $DNS1" /etc/resolv.conf &&
+  if [ -n "$DNS1" ] && ! grep -q "^nameserver $DNS1" /etc/resolv-peerdns.conf &&
    tr=$(mktemp /tmp/XXXXXX) ; then
     current_replacement="$DNS1"
     next_replacement="$DNS2"
     search=
-    (cat /etc/resolv.conf ; echo EOF ; echo EOF) | while read answer ; do
+    (cat /etc/resolv-peerdns.conf ; echo EOF ; echo EOF) | while read answer ; do
 	case $answer in
 	    nameserver*|EOF)
 		if [ -n "$current_replacement" ] ; then
diff -up initscripts-9.49.24/sysconfig/network-scripts/network-functions.peerdns initscripts-9.49.24/sysconfig/network-scripts/network-functions
--- initscripts-9.49.24/sysconfig/network-scripts/network-functions.peerdns	2015-01-15 01:57:03.000000000 -0700
+++ initscripts-9.49.24/sysconfig/network-scripts/network-functions	2015-04-03 09:55:58.317658970 -0600
@@ -364,11 +364,11 @@ need_hostname ()
 set_hostname ()
 {
     hostname $1
-    if ! grep search /etc/resolv.conf >/dev/null 2>&1; then
+    if ! grep search /etc/resolv-peerdns.conf >/dev/null 2>&1; then
 	domain=$(echo $1 | sed 's/^[^\.]*\.//')
 	if [ -n "$domain" ]; then
 		rsctmp=$(mktemp /tmp/XXXXXX);
-		cat /etc/resolv.conf > $rsctmp
+		cat /etc/resolv-peerdns.conf > $rsctmp
 		echo "search $domain" >> $rsctmp
 		change_resolv_conf $rsctmp
 		/bin/rm -f $rsctmp
@@ -528,10 +528,10 @@ is_bonding_device ()
    [ -f "/sys/class/net/$1/bonding/slaves" ]
 }
 
-# Invoke this when /etc/resolv.conf has changed:
+# Invoke this when /etc/resolv-peerdns.conf has changed:
 change_resolv_conf ()
 {
-    s=$(/bin/grep '^[\ \	]*option' /etc/resolv.conf 2>/dev/null);
+    s=$(/bin/grep '^[\ \	]*option' /etc/resolv-peerdns.conf 2>/dev/null);
     if [ "x$s" != "x" ]; then
        s="$s"$'\n';
     fi;
@@ -558,11 +558,11 @@ change_resolv_conf ()
 	  s=$(cat $1);
        fi;
     fi;
-    (echo "$s" > /etc/resolv.conf;) >/dev/null 2>&1;
+    (echo "$s" > /etc/resolv-peerdns.conf;) >/dev/null 2>&1;
     r=$?
     if [ $r -eq 0 ]; then
-	[ -x /sbin/restorecon ] && /sbin/restorecon /etc/resolv.conf >/dev/null 2>&1 # reset the correct context
-	/usr/bin/logger -p local7.notice -t "NET" -i "$0 : updated /etc/resolv.conf";
+	[ -x /sbin/restorecon ] && /sbin/restorecon /etc/resolv-peerdns.conf >/dev/null 2>&1 # reset the correct context
+	/usr/bin/logger -p local7.notice -t "NET" -i "$0 : updated /etc/resolv-peerdns.conf";
 	[ -e /var/run/nscd/socket ] && /usr/sbin/nscd -i hosts; # invalidate cache
     fi;
     return $r;
diff -up initscripts-9.49.24/sysconfig.txt.peerdns initscripts-9.49.24/sysconfig.txt
--- initscripts-9.49.24/sysconfig.txt.peerdns	2015-01-15 01:57:03.000000000 -0700
+++ initscripts-9.49.24/sysconfig.txt	2015-04-03 09:55:58.322658983 -0600
@@ -517,10 +517,10 @@ Files in /etc/sysconfig/network-scripts/
     WINDOW=
       Default window for routes from this device
     PEERDNS=yes|no
-      modify /etc/resolv.conf if peer uses msdns extension (PPP only) or
+      modify /etc/resolv-peerdns.conf if peer uses msdns extension (PPP only) or
       DNS{1,2} are set, or if using dhclient. default to "yes".
     DNS{1,2}=<ip address>
-      provide DNS addresses that are dropped into the resolv.conf
+      provide DNS addresses that are dropped into the resolv-peerdns.conf
       file if PEERDNS is not set to "no".
     SCOPE=
       Set to "scope SCOPE-ID" to set a non-default scope for a statically
