diff -rup initscripts-9.49.37/sysconfig/network-scripts/ifup-eth initscripts-9.49.37-multiwan/sysconfig/network-scripts/ifup-eth
--- initscripts-9.49.37/sysconfig/network-scripts/ifup-eth	2016-09-12 06:47:53.000000000 -0400
+++ initscripts-9.49.37-multiwan/sysconfig/network-scripts/ifup-eth	2016-11-29 14:56:13.262415740 -0500
@@ -279,6 +279,19 @@ else
         ip addr change ${ipaddr[$idx]}/${prefix[$idx]} dev ${REALDEVICE} valid_lft forever preferred_lft forever
     done
 
+    # ClearFoundation -- improve default route handling in multi-WAN environment
+    if [ -e /etc/clearos/network.conf ]; then
+        . /etc/clearos/network.conf
+    fi
+    MULTIEXT=`echo $EXTIF | awk '{ print $2 }'`
+    if [ -n "$MULTIEXT" ]; then
+        DEFAULTEXISTS=`ip -o route | grep ^default`
+        if [ -n "$DEFAULTEXISTS" ]; then
+            DEFROUTE="no"
+        fi
+    fi
+    # ClearFoundation -- end
+
     # Set a default route.
     if [ "${DEFROUTE}" != "no" ] && [ -z "${GATEWAYDEV}" -o "${GATEWAYDEV}" = "${REALDEVICE}" ]; then
 	# set up default gateway. replace if one already exists
