diff -up initscripts-9.49.24/sysconfig/network-scripts/ifup-eth.multiwan initscripts-9.49.24/sysconfig/network-scripts/ifup-eth
--- initscripts-9.49.24/sysconfig/network-scripts/ifup-eth.multiwan	2015-01-15 01:57:03.000000000 -0700
+++ initscripts-9.49.24/sysconfig/network-scripts/ifup-eth	2015-04-03 09:55:58.297658916 -0600
@@ -274,6 +274,19 @@ else 
         fi
     done
 
+    # ClearFoundation -- improve default route handling in multi-WAN environment
+    if [ -e /etc/clearos/network.conf ]; then
+        . /etc/clearos/network.conf
+    fi
+    MULTIEXT=`echo $EXTIF | awk '{ print $2 }'`
+    if [ -n "$MULTIEXT" ]; then
+        DEFAULTEXISTS=`ip -o route | grep ^default`
+        if [ -n "$DEFAULTEXISTS" ]; then
+            DEFROUTE="no"
+        fi
+    fi
+    # ClearFoundation -- end
+
     # Set a default route.
     if [ "${DEFROUTE}" != "no" ] && [ -z "${GATEWAYDEV}" -o "${GATEWAYDEV}" = "${REALDEVICE}" ]; then
 	# set up default gateway. replace if one already exists
diff -up initscripts-9.49.24/sysconfig/network-scripts/ifup-ppp.multiwan initscripts-9.49.24/sysconfig/network-scripts/ifup-ppp
--- initscripts-9.49.24/sysconfig/network-scripts/ifup-ppp.multiwan	2015-01-15 01:57:03.000000000 -0700
+++ initscripts-9.49.24/sysconfig/network-scripts/ifup-ppp	2015-04-03 09:55:58.300658924 -0600
@@ -52,6 +52,8 @@ fi
 # check that xDSL connection
 if [ "$TYPE" = "xDSL" ] ; then
     if [ -x /usr/sbin/adsl-start ] ; then
+        # ClearFoundation - multiple PPPoE interfaces need to behave
+        PPPD_EXTRA="unit $(echo -n "${DEVICE}" | sed -e 's/[[:alpha:]]*//g') $PPPD_EXTRA" \
         adsl-start /etc/sysconfig/network-scripts/$CONFIG
         exit $?
     else
@@ -99,6 +101,18 @@ fi
 if [ "${ESCAPECHARS}" != yes ] ; then
   opts="$opts asyncmap 00000000"
 fi
+# ClearFoundation -- Multi-WAN friendly default route policy
+if [ -e /etc/clearos/network.conf ]; then
+  . /etc/clearos/network.conf
+fi
+MULTIEXT=`echo $EXTIF | awk '{ print $2 }'`
+if [ -n "${MULTIEXT}" ]; then
+  DEFAULTEXISTS=`ip route list | awk '/^default / { print $3 }'`
+  if [ -n "$DEFAULTEXISTS" ]; then
+    DEFROUTE="no"
+  fi 
+fi
+# ClearFoundation - end
 if [ "${DEFROUTE}" != no ] ; then
   # pppd will no longer delete an existing default route
   # so we have to help it out a little here.
